Marlin/Marlin_main.cpp

```diff
@@ -4166,46 +4166,20 @@ inline void gcode_M105() {
    * M106: Set Fan Speed
    */
 
-  // 127 (50%) is maxy duty cycle for 12V fans
-  #define MAX_FAN_DUTY  127
-
   inline void gcode_M106() {
     // Desired speed given
-    if ((code_seen('S')) && (code_value() <= MAX_FAN_DUTY)) {
-      fanSpeed = code_value();
-    }
-    // No speed given, defaults to MAX_FAN_DUTY
-    else {
-      fanSpeed = MAX_FAN_DUTY;
+    if ((code_seen('S'))) {
+      fanSpeed = (uint8_t)code_value();
     }
 
-    Wire.beginTransmission(CART_HOLDER_ADDR);
-    Wire.write(SET_FAN_DRIVE_0_PWM);
-    Wire.write(fanSpeed);
-    Wire.endTransmission();
-
-    #if defined(DEBUG)
-      SERIAL_PROTOCOLLNPGM("Command: 'Set Fan Speed' Sent");
-      SERIAL_PROTOCOL("fanSpeed = ");
-      SERIAL_PROTOCOL(fanSpeed);
-    #endif // end DEBUG
+    I2C__SetFanDrive0PWM(fanSpeed);
```

Comment:
fanSpeed is actually defined as 0 as a global, which is pretty lame in my estimations. Should I refactor this into a local variable?